Canada: make observed New Year's Day fall on or after the holiday

jasonjensen opened the pull request:
The Canadian rules are ambiguous w.r.t. where to place the observed day when a statutory holiday falls on a weekend. According to Wikipedia the observed day can come before or after the holiday:
https://en.wikipedia.org/wiki/Public_holidays_in_Canada#Holidays_occurring_on_non-work_days

However, in the case of NYD, this has historically been falling after new year's. Since 1980 NYD has fallen on a Saturday six times (1983, 1994, 2000, 2005, 2011, and 2022). In all cases, the TSX (Toronto Stock Exchange) has had data on December 31st (the Friday) and no data on January 3rd (the following Monday).

Similarly, NYD has fallen on a Sunday seven times (1978, 1984, 1989, 1995, 2006, 2012, 2017). A similar pattern is observed there; data on Dec 30th and no data on Jan 2nd. For national markets at least, there is consistency in the placement of NYD.

arkid15r reviewed (changes requested):
This [test](https://github.com/dr-prodigy/python-holidays/blob/2cc3151877b59a0344f7d241ea63515d60ed6d04/test/countries/test_canada.py#L24) will fail. Please test your changes locally (see [CONTRIBUTING.rst](https://github.com/dr-prodigy/python-holidays/blob/master/CONTRIBUTING.rst))

```
$ python -m pytest .
# if you want to retrieve uncovered lines too:
$ python -m pytest --cov-report term-missing .
```

jasonjensen commented on holidays/countries/canada.py:
```diff
@@ -77,10 +78,8 @@ def _populate(self, year):
-        if self.observed and date(year, DEC, 31).weekday() == FRI:
-            self[date(year, DEC, 31)] = name + " (Observed)"
+        elif self.observed and date(year, JAN, 1).weekday() == SAT:
+            self[date(year, JAN, 1) + rd(days=+2)] = name + " (Observed)"
```
Thanks!

jasonjensen commented:
> This [test](https://github.com/dr-prodigy/python-holidays/blob/2cc3151877b59a0344f7d241ea63515d60ed6d04/test/countries/test_canada.py#L24) will fail. Please test your changes locally (see [CONTRIBUTING.rst](https://github.com/dr-prodigy/python-holidays/blob/master/CONTRIBUTING.rst))

My apologies. I have now adjusted the test for Canada and run the tests locally (albeit only on python 3.9). I also incorporated @KJhellico 's suggestion for improving the code.

dr-prodigy commented:
Merged on beta branch.. thank you very much! 👍
